### DISCUSSION TITLE
Tree onNodeUnselect should reselect the node, avoid default behaviour

### DISCUSSION THREAD

**Discussion by MehdiRaza1998** [primefaces/community#2761] [PrimeNG]
I am using Prime ng tree for tree render in my application. 
Usecase: Simply, when node is selected again (onNodeUnselect) , I want the node to be still considered as selected. So, I can call the method of NodeSelect again and retrieve the fresh data for the node. 

```
<p-tree
        [value]="getValidTreeNodeItems(treeObj?.treeData)"
        selectionMode="single"
        (onNodeSelect)="onNodeSelect($event)"
        (onNodeUnselect)="preventUnselect($event)" > ....
//prevent 
 preventUnselect(event: any) {
        event.originalEvent.preventDefault(); //doesn't work
        event.originalEvent.stopPropagation(); //doesn't work either
```
        
thus the node is considered unselected, and all the selection classes are removed too. 
I tried the way of applying classes (like p-highlight) to the node which just got unselected to make it look selected, but this leads to maintaining another logical stuff to assign and remove the classes, which I want to avoid. 

Thanks in advance for the help.

**Comment by Ironmunkii** [comment]
I need the answer to this as well...

I tried what you did @MehdiRaza1998 too and it doesn't work. 

Manually setting the active node doesn't work either.

**Comment by rfreydi** [comment]
By reading the codesource, I find a way that should not imply side effect:

```html
<!-- `selection` is a Signal but it can be a classic property -->
<p-tree 
  #treeInstance
  [selection]="selection()" 
  (onNodeSelect)="selection.set($event.node)"
  (onNodeUnselect)="treeInstance.selection = $event.node"
/>
```

The only downside: the `selectionChange` event will still be fired but with the value selected (instead of `null`) so keep that in mind
